doc/source/ray-core/examples/automl_for_time_series.ipynb

```diff
@@ -0,0 +1,616 @@
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ray.init(ignore_reinit_error=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will break up our logic into several functions and a Ray [task](ray-remote-functions).\n",
+    "\n",
+    "The Ray task is `train_and_evaluate_fold`, which contains all the logic necessary to fit and evaluate a model on a CV fold of data. We structure our task to take in a dataset and indices splitting it into train and test - that way, we can keep one instance of the dataset in the Ray object store and split it in each task separately. We are defining this as a Ray task as we want all folds to be evaluated in parallel on a Ray cluster - Ray will handle all orchestration and execution. Each task will reserve 1 CPU core by default."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@ray.remote\n",
+    "def train_and_evaluate_fold(\n",
+    "    model: _TS,\n",
+    "    df: pd.DataFrame,\n",
+    "    train_indices: np.ndarray,\n",
+    "    test_indices: np.ndarray,\n",
+    "    label_column: str,\n",
+    "    metrics: Dict[str, Callable[[pd.Series, pd.Series], float]],\n",
+    "    freq: str = \"D\",\n",
+    ") -> Dict[str, float]:\n",
+    "    try:\n",
+    "        # Create the StatsForecast object with train data & model.\n",
+    "        statsforecast = StatsForecast(\n",
+    "            df=df.iloc[train_indices], models=[model], freq=freq\n",
+    "        )\n",
+    "        # Make a forecast and calculate metrics on test data.\n",
+    "        # This will fit the model first automatically.\n",
+    "        forecast = statsforecast.forecast(len(test_indices))\n",
+    "        return {\n",
+    "            metric_name: metric(\n",
+    "                df.iloc[test_indices][label_column], forecast[model.__class__.__name__]\n",
+    "            )\n",
+    "            for metric_name, metric in metrics.items()\n",
+    "        }\n",
+    "    except Exception:\n",
+    "        # In case the model fit or eval fails, return None for all metrics.\n",
+    "        return {metric_name: None for metric_name, metric in metrics.items()}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`evaluate_models_with_cv` is a driver function to run our optimization loop. We take in a list of models (with their parameters already set) and the dataframe.\n",
+    "\n",
+    "The dataframe is put into the Ray object store and reused, which means we only need to serialize it once. That way, we avoid an {doc}`/ray-core/patterns/pass-large-arg-by-value`.\n",
```

Comment:
Don't link to docs. Use internal refs.

Reply:
Curious, why is that?

Reply:
If you move files around or move any sections away from the file, the links don't break. It's much more resilient to any slight refactoring.

Reply:
I feel moving the file is similar to renaming the internal ref and is caught by our CI for any broken links. The fix is simple enough to be a search and replace.

I do like that when I link to a doc, sphinx will include the doc title...